substrate/frame/support/src/storage/types/value.rs

```diff
@@ -30,11 +30,29 @@ use sp_arithmetic::traits::SaturatedConversion;
 use sp_metadata_ir::{StorageEntryMetadataIR, StorageEntryTypeIR};
 use sp_std::prelude::*;
 
-/// A type that allow to store a value.
+/// A type representing a *value* in storage. A *storage value* is a single value of a given type
+/// stored on-chain.
 ///
-/// Each value is stored at:
-/// ```nocompile
-/// Twox128(Prefix::pallet_prefix()) ++ Twox128(Prefix::STORAGE_PREFIX)
+/// For general information regarding the `#[pallet::storage]` attribute, refer to 
+/// [`crate::pallet_macros::storage`].
+/// 
+/// # Example
+/// 
+/// ```
+/// #[frame_support::pallet]
+/// mod pallet {
+///     # use frame_support::pallet_prelude::*;
+///     # #[pallet::config]
+///     # pub trait Config: frame_system::Config {}
+///     # #[pallet::pallet]
+///     # pub struct Pallet<T>(_);
+/// 	/// A kitchen-sink storage value, with all possible additional attributes.
+///     #[pallet::storage]
+/// 	#[pallet::getter(fn foo)]
+/// 	#[pallet::storage_prefix = "OtherFoo"]
+/// 	#[pallet::unbounded]
+///     pub type Foo<T> = StorageValue<_, u32, ValueQuery>;
+/// }
```

Comment:
another idea is for you to add a single test/snippet here to each storage type where you re-create the storage key to demonstrate the point. Something like:

```
fn main() {
	let expected_key = twox(pallet_name) ++ twox(Foo)
	let key = Foo::hashed_key();
	assert_eq!(expected., key);
}
```

Sadly would require you to build a runtime, but thanks to all the recent work it is only a few lines of code now :)